packages/repository/src/relations/relation.types.ts

```diff
@@ -87,12 +87,12 @@ export interface HasManyThroughDefinition extends RelationDefinitionBase {
   /**
    * The foreign key in the source model, e.g. Customer#id.
    */
-  keyFrom: string;
+  keyFrom?: string;
```

Comment:
Why make these keys optional? If justified, this change should be PR of its own to minimize the size of the PR.

Reply:
I'll happy to extract this into a PR if there is agreement in overall structure.

Reply:
I am ok with these changes, I see they are consistent with what we already have in place for `HasManyDefinition` (see above in the source code file).

I agree a new PR would allow us to land these fixes sooner, on the other hand such PR would be a bit anemic - since no code is using `HasManyThroughDefinition` yet, there will be no way how to verify correctness of such small PR.

In that light, I think it may be better better to keep these changes inside this bigger PR.

Reply:
Cross-posting from https://github.com/strongloop/loopback-next/pull/4438#discussion_r402117145:

> I think it would be best to extract the changes in relation definitions, the implementation bits of resolving through relations and the accompanying unit tests into a new pull request, to get a smaller patch that's easier to review & iterate on.

I think this is the best way how to get a PR changing relation definition that's smaller but still meaningful.